pkg/proxy/iptables/proxier.go

```diff
@@ -1201,74 +1188,69 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 		}
 
-		// Capture load-balancer ingress.
+		// Capture load-balancer ingress. This applies only to load balancers that
+		// forward traffic to the node with the destination IP still set to the IP
+		// of the load balancer (preserving the original source IP and allowing
+		// Direct Server Return). Load balancers that don't behave that way won't
+		// set the load balancer IPs in the service status, and instead just forward
+		// the traffic to a NodePort, so the handling below (including, in particular,
+		// the implementation of LoadBalancerSourceRanges) will be skipped.
 		fwChain := svcInfo.serviceFirewallChainName
 		for _, ingress := range svcInfo.LoadBalancerIPStrings() {
-			if hasEndpoints {
+			targetChainFromKubeServices := extTargetChain
+
+			// We need a FW chain if loadBalancerSourceRanges is used
+			if hasAnyEndpoints && len(svcInfo.LoadBalancerSourceRanges()) > 0 {
```

Comment:
I think the `len(svcInfo.LoadBalancerSourceRanges()) > 0` check here is new but some of the new rules added here should apply when the source range is not set right?

Reply:
Ah I think I follow now, some of the other rules were removed in this if block and moved below under another one.